tests/conftest.py

```diff
@@ -81,3 +83,21 @@ def tensor_files(tmp_path, tensors):
         torch.save(t, path)
         paths.append(path)
     return iter(paths)
+
+
+@pytest.fixture(scope="session")
+def images(tensors):
+    return [Image.fromarray((t * np.iinfo(np.uint16).max).squeeze().numpy().astype(np.uint16)) for t in tensors]
+
+
+@pytest.fixture
+def image_input(images):
+    return iter(images)
+
+
+@pytest.fixture(scope="session")
+def image_files(tmpdir_factory, images):
+    tmp_path = tmpdir_factory.mktemp("img_data")
+    paths = [Path(tmp_path / f"image_{i}.png") for i, img in enumerate(images)]
+    [img.save(path) for img, path in zip(images, paths)]
```

Comment:
Since this list comprehension isn't actually used, maybe this code could be rewritten to loop through `images` only once? Not a big deal either way.
```python
tmp_path = Path(tmpdir_factory.mktemp("img_data"))
for i, image in enumerate(images):
    path = tmp_path / f"image_{i}.png"
    image.save(path)
    yield path
```

Reply:
I adopted most of this change. I needed to keep the return statement as written. I was getting test failures as a `yield`, perhaps because `yield` has special meaning in a pytest fixture.